lib/Conversion/TritonGPUToLLVM/TypeConverter.cpp

```diff
@@ -105,23 +68,14 @@ Type TritonGPUToLLVMTypeConverter::convertTritonTensorType(
 Type TritonGPUToLLVMTypeConverter::convertMemDescType(
     MemDescType type, const TargetInfoBase &targetInfo) {
   auto ctx = type.getContext();
-  Attribute layout = type.getEncoding();
-  SmallVector<int64_t> shape(type.getShape().begin(), type.getShape().end());
-  SmallVector<Type, 4> types;
-  // base ptr
-  auto ptrType =
-      LLVM::LLVMPointerType::get(ctx, targetInfo.getSharedAddressSpace());
-  types.push_back(ptrType);
-  // shape dims
-  auto rank = type.getShape().size();
-  // offsets
-  for (auto i = 0; i < rank; i++) {
-    types.push_back(IntegerType::get(ctx, 32));
+  if (isa<SharedEncodingAttr>(type.getEncoding())) {
+    return getSharedMemoryType(ctx, type.getRank(), targetInfo);
   }
-  return LLVM::LLVMStructType::getLiteral(ctx, types);
+  emitError(UnknownLoc::get(ctx), "unsupported MemDescType encoding");
```

Comment:
Let's me if you have a better solution for other encodings used by MemDesc. @ThomasRaoux

Reply:
We don't convert MemDesc any differently for the non-smem variant. It would be easier if you just kept them going through the same path to avoid creating new merge issues.

Reply:
Shall I use the same name `getSharedMemoryType`? Not sure if it's a bit confusing